src/index.js

```diff
@@ -6,12 +6,68 @@ const UUID = require('keycloak-connect/uuid');
 const Boom = require('boom');
 const _ = require('lodash');
 const pkg = require('../package.json');
-const axios = require('axios');
+const crypto = require('crypto');
 const urljoin = require('url-join');
 
 const getProtocol = (request) => request.headers['x-forwarded-proto'] || request.server.info.protocol;
 const getHost = (request) => request.headers['x-forwarded-host'] || request.info.host;
 
+class ActionTokenVerifier {
+
+    constructor(grantManager) {
+        this.grantManager = grantManager;
+    }
+
+    async verify(token, {action, resource}) {
+
+        const tokenRules = [
+            {
+                verify: () => token,
+                error: new Error('Invalid token (missing)')
```

Comment:
I guess it would be better not to create errors and rules in the verify method.
Consider extractin tokenRules outside of this class or at least putting them in constructor.
Make functions out of `error: new Error('Invalid token (missing)')` constructs.